terraform/modules/rds/main.tf

```
@@ -0,0 +1,56 @@
+# Non-compliant RDS configuration (missing encryption, backups, and monitoring)
+resource "aws_db_instance" "main" {
+  identifier        = "${var.environment}-db"
```

The RDS instance is not compliant with SOC2 requirements due to several issues: storage encryption is disabled, backup retention is set to 0, and monitoring is not enabled. Additionally, the instance is publicly accessible, which is a security risk.

```suggestion:-0+0
resource "aws_db_instance" "main" {
  identifier        = "${var.environment}-db"
  engine            = "postgres"
  engine_version    = "13.7"
  instance_class    = var.instance_class
  allocated_storage = 20

  name              = var.db_name
  username          = var.db_username
  password          = var.db_password

  publicly_accessible = false

  storage_encrypted = true

  backup_retention_period = 30
  skip_final_snapshot     = false

  monitoring_interval = 60

  vpc_security_group_ids = [aws_security_group.db.id]

  tags = {
    Name        = "${var.environment}-db"
    Environment = var.environment
  }
}
```